network/p2p/dns/resolver_test.go

```diff
@@ -201,6 +202,59 @@ func TestResolver_Expired_Invalidated(t *testing.T) {
 	require.Zero(t, txtSize)
 }
 
+// TestResolver_OverloadQueue initializes a resolver with worker = 1 and queue size = 2 for both IPLookup and TXTLookup
+// Test will then attempt to overload the queue to test that queries outside the queue size would be dropped but resolver does not fail.
+func TestResolver_OverloadQueue(t *testing.T) {
+	ipQueryCount := 0
+	txtQueryCount := 0
+	basicResolver := mocknetwork.BasicResolver{}
+	dnsCache := herocache.NewDNSCache(
+		DefaultCacheSize,
+		unittest.Logger(),
+		metrics.NewNoopCollector(),
+		metrics.NewNoopCollector(),
+	)
+
+	resolver := NewResolver(
+		unittest.Logger(),
+		metrics.NewNoopCollector(),
+		dnsCache,
+		WithBasicResolver(&basicResolver),
+		WithTTL(1*time.Second),
+		WithResolverSize(1, 1, 2, 2))
+
+	cancelCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx, _ := irrecoverable.WithSignaler(cancelCtx)
+	resolver.Start(ctx)
+	unittest.RequireCloseBefore(t, resolver.Ready(), 100*time.Millisecond, "could not start dns resolver on time")
+
+	size := 10 // 10 text and 10 ip domains.
+	times := 1 // each domain is queried 5 times.
+	txtTestCases := testnetwork.TxtLookupFixture(size)
+	ipTestCases := testnetwork.IpLookupFixture(size)
+
+	// each domain is resolved only once through the underlying resolver, and then is cached for subsequent times.
+	mockBasicResolverForDomains(t, &basicResolver, ipTestCases, txtTestCases, happyPath, 2, LogResolveQueryCounts(&ipQueryCount, &txtQueryCount))
+	queryWG := syncThenAsyncQuery(t, times, resolver, txtTestCases, ipTestCases, happyPath)
+	unittest.RequireReturnsBefore(t, queryWG.Wait, 1*time.Second, "could not perform all queries on time")
+
+	time.Sleep(2 * time.Second) // waits enough for cache to get invalidated
+
+	queryWG = syncThenAsyncQuery(t, times, resolver, txtTestCases, ipTestCases, happyPath)
+
+	unittest.RequireReturnsBefore(t, queryWG.Wait, 1*time.Second, "could not perform all queries on time")
+	//Verifying
+	cancel()
+	unittest.RequireCloseBefore(t, resolver.Done(), 100*time.Millisecond, "could not stop dns resolver on time")
+	log.Printf("ipQueryCount: %d txtQueryCount: %d", ipQueryCount, txtQueryCount)
```

Comment:
Please use `unittest.Logger()` for logging in the tests, it is more consistent with our codebase, also we can enable and disable it by adding or removing `-vv` flag, respectively. Thanks.